### ISSUE TITLE
Importing existing SDDCs does not work correctly

### ISSUE THREAD

**Issue by mattycourtney** [vmware/terraform-provider-vmc#56]
I want to use `terraform import` to manage an existing SDDC, but hit some unexpected behavior. 
It seems that when doing an import, some attributes that are written to `terraform.tfstate` do not reflect the current state of the SDDC.

# Steps to reproduce:
1. Create a `main.tf` file like the example below
```
provider "vmc" {
	refresh_token = "YOUR_REFRESH_TOKEN_HERE"
	org_id = "YOUR_ORG_ID_HERE"
}

resource "vmc_sddc" "SDDC_1" {
	sddc_name = "SDDC_1"
	region = "AP_SOUTHEAST_2"
	num_host = 3
}
```
2. Run `terraform import vmc_sddc.SDDC_1 sddc_id_here`
3. Run `terraform plan`
 
 
# Expected result:
 Terraform reports no changes required
 
 
# Actual result:
 Terraform reports the SDDC must be replaced, because six attributes force the replacement 

The attributes are `delay_account_link`, `deployment_type`, `provider_type`, `region`, `skip_creating_vxlan`, `sso_domain`.
 
 
# Manual workaround
I was able to edit the `terraform.tfstate` file by hand and replace the values that were set incorrectly.

Old Value | New Value
---------- | ----------
"delay_account_link": null | "delay_account_link": false
"deployment_type": null | "deployment_type": "SingleAZ"
"provider_type": null | "provider_type": "AWS"
"region": null | "region": "AP_SOUTHEAST_2"
"skip_creating_vxlan": null | "skip_creating_vxlan": true
"sso_domain": null | "sso_domain": "vmc.local"

After editing those values, `terraform plan` reported no changes required (as expected).

# Version affected
Terraform v0.13.0 + provider registry.terraform.io/terraform-providers/vmc v1.3.0

**Comment by mattycourtney** [comment]
My colleague @nvibert suggested that I need to include the above mentioned attributes in my `main.tf` prior to running `terraform import`. I tried that (see example `main.tf` below) and the results were the same. 

```
provider "vmc" {
        refresh_token = "REDACTED"
        org_id = "REDACTED"
}

resource "vmc_sddc" "SDDC_1" {
        sddc_name = "SDCC_1"
        region = "AP_SOUTHEAST_2"
        num_host = 3
        delay_account_link = false
        deployment_type = "SingleAZ"
        provider_type = "AWS"
        skip_creating_vxlan = true
        sso_domain = "vmc.local"
}
```

**Comment by saloneerege** [comment]
Thanks for the update @mattycourtney. I will look into this issue.

**Comment by uprightvinyl** [comment]
I'm wondering if there has been any progress on this issue? I can report the exact same behaviour.

**Comment by saloneerege** [comment]
@uprightvinyl thanks for reporting this issue. We will prioritize this issue.

**Comment by shylasrinivas** [comment]
@mattycourtney can you follow exact steps provided in [this](https://learn.hashicorp.com/tutorials/terraform/state-import?in=terraform/state&utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS#import-the-container-into-terraform) tutorial and let me know the behavior in your setup? 
I started with an empty sddc resource definition in main.tf, imported an existing SDDC and then cherry-picked required arguments into config. 

Initial config
```
terraform {
  required_providers {
    vmc = {
      source = "vmware/vmc"
      version = "1.4.0"
    }
  }
}

provider "vmc" {
  # Configuration options
  refresh_token = var.api_token
  org_id        = var.org_id
}

resource "vmc_sddc" "sddc_i" {
}

```
Config after import
```
terraform {
  required_providers {
    vmc = {
      source = "vmware/vmc"
      version = "1.4.0"
    }
  }
}

provider "vmc" {
  # Configuration options
  refresh_token = var.api_token
  org_id        = var.org_id
}

resource "vmc_sddc" "sddc_i" {
 sddc_name = "sddc_i"
 region = "val1"
 num_host = intval
 sddc_type  = "val2"
 vpc_cidr = "val3"
 vxlan_subnet = "val4"
}
```

delay_account_link is set during create internally, and is not returned as part get API call. Hence it is added as part of plan. skip_creating_vxlan for some reason is marked to change from false to true which I am verifying.

**Comment by mattycourtney** [comment]
@shylasrinivas I no longer work at VMware or have access to a VMC environment. @nvibert is a keen Terraform user on the VMC SE team, you could try asking if he is willing and able to test this for you.

**Comment by shylasrinivas** [comment]
@mattycourtney @nvibert @uprightvinyl Upon further testing, we have identified limitations with existing import SDDC functionality. 

`terraform import` command works as expected and using below config post import for `terraform plan` shows that SDDC must be replaced. This will destroy the existing SDDC and create a new one which is undesirable. It is due to `delay_account_link` argument which is not returned as part of Get API call and is set as ForceNew in the code since there is no API to update it. We have reported this to the VMC API team and there is nothing much we can do till the API is available.

```
resource "vmc_sddc" "sddc_i" {
 sddc_name = "test_sddc"
 region = "US_EAST_1"
 num_host = 1
 delay_account_link  = false
 skip_creating_vxlan  = false
 sddc_type  = "1NODE"
 vpc_cidr  = "10.2.0.0/16"
 vxlan_subnet = "192.168.1.0/24"
}
```
